test/governance/ATMTokenTest.js

```diff
@@ -0,0 +1,190 @@
+// JS Libraries
+const withData = require('leche').withData;
+const { t, NULL_ADDRESS  } = require('../utils/consts');
+const Timer = require('../../scripts/utils/Timer');
+
+// Smart contracts
+const ATMToken = artifacts.require("./ATMToken.sol");
+
+contract('ATMTokenTest', function (accounts) {
+    let instance;
+    const daoAgent = accounts[0];
+    const daoMember1 = accounts[2];
+    const daoMember2 = accounts[3];
+    const timer = new Timer(web3);
+
+    beforeEach('Setup for each test', async () => {
+        instance = await ATMToken.new(10000);
+    });
+
+    withData({
+        _1_mint_no_vesting_basic: [daoMember1, 2000, undefined, false],
+        _2_mint_no_vesting_above_cap: [daoMember1, 11000, 'ERC20Capped: cap exceeded', true],
+        _3_mint_no_vesting_zero_address: [NULL_ADDRESS, 3000, "ERC20: mint to the zero address", true]
+    },function(
+        receipent,
+        amount,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to mint correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                const result = await instance.mint(receipent, amount, { from: daoAgent });
+                // Assertions
+                assert(!mustFail, 'It should have failed because the amount is greater than the cap');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _4_mint_vesting_basic: [daoMember2, 1000, 7000, undefined, false],
+        _5_mint_vesting_above_cap: [daoMember2, 21000, 7000, 'ERC20Capped: cap exceeded', true],
+        _6_mint_vesting_zero_address: [NULL_ADDRESS, 3000, 60000, "ERC20: mint to the zero address", true],
+        _7_mint_vesting_incorrect_address: [daoMember1, 1000, 7000, 'Account does not have a vesting balance!', true],
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to mint correctly', mustFail), async function() {
```

Comment:
Please, verify if the function "mint" matches the function you are testing.

Reply:
Verified